### PR TITLE
pythonPackages.pyglet: 1.4.2 -> 1.5.19

### PR DESCRIPTION
**Pull request by tpwrules**
###### Motivation for this change
Upgrade to a recent version of pyglet.

###### Things done
Updated the list of patched library paths for new features. I tried to enable tests, but even though headless mode is now present they can't be run during a build anyway.

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- X ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
Assume you've tried `xvfb`

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 134735` at d93f8e57 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built successfully:</summary>
  <ul>
    <li>printrun</li>
    <li>python38Packages.moderngl-window</li>
    <li>python38Packages.pyglet</li>
    <li>python38Packages.pytmx</li>
    <li>python39Packages.moderngl-window</li>
    <li>python39Packages.pyglet</li>
    <li>python39Packages.pytmx</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/pyglet/default.nix:98:0:
```
   |
98 |     description = "A cross-platform windowing and multimedia library";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/pyglet/default.nix:96:3:
```
   |
96 |   meta = with lib; {
   |   ^
```
</li>
  </ul>
</details>

**Comment by tpwrules** [comment]
No, `xvfb` does not help. I tried it, but the problem is a lack of GL, not a lack of X11.

File: pkgs/development/python-modules/pyglet/default.nix [hunk 1]
```diff
@@ -1,26 +1,31 @@
...
 , ffmpeg-full
+, openal
+, libpulseaudio
 }:
 
 buildPythonPackage rec {
-  version = "1.4.2";
+  version = "1.5.19";
   pname = "pyglet";
+  disabled = !isPy3k;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyglet/default.nix:23]: Please use pythonOlder.

File: pkgs/development/python-modules/pyglet/default.nix [hunk 2]
```diff
@@ -55,26 +62,39 @@ buildPythonPackage rec {
...
   doCheck = false;
 
   checkInputs = [
     pytest
   ];
 
+  # test list taken from .travis.yml
   checkPhase = ''
-    py.test tests/unit tests/integration
+    PYGLET_HEADLESS=True pytest --ignore=tests/base --ignore=tests/interactive --ignore=tests/integration --ignore=tests/unit/text/test_layout.py
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyglet/default.nix:93]: Please use pytestCheckHook.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pyglet/default.nix (hunk 2 above)
Comment (tpwrules) [reply pkgs/development/python-modules/pyglet/default.nix:93]: How do I pass the environment variable PYGLET_HEADLESS with pytestCheckHook?

Comment (tpwrules) [reply pkgs/development/python-modules/pyglet/default.nix]: Figured this out, thanks for the suggestion.

File: pkgs/development/python-modules/pyglet/default.nix (hunk 1 above)
Comment (tpwrules) [reply pkgs/development/python-modules/pyglet/default.nix]: Done, documentation says <3.6 is not supported.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 134735` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
7 packages built:
</summary>
<ul>
<li>printrun</li>
<li>python38Packages.moderngl-window</li>
<li>python38Packages.pyglet</li>
<li>python38Packages.pytmx</li>
<li>python39Packages.moderngl-window</li>
<li>python39Packages.pyglet</li>
<li>python39Packages.pytmx</li>
</ul>
</details>

**Pull request merged**